File: neo/frontend/mockdata/admin/workspace.ts
// ============================================================
// Types
// ============================================================
import type { WorkspaceStatus } from "@/components/workspace/workspace-types";
import type { Workspace as WorkspaceBase } from "@/components/workspace/workspace-types";

// Re-export for convenience
export type { WorkspaceStatus };
export type Workspace = WorkspaceBase;

// ============================================================
// Workspaces List
// ============================================================
export const mockWorkspaces: Workspace[] = [
  {
    id: 1,
    name: "CRM 工作区",
    code: "crm-workspace",
    description: "客户关系管理团队的工作区",
    status: "active",
    org_id: 1,
    owner_id: 1,
    member_count: 15,
    project_count: 8,
    created_at: "2026-05-01T10:00:00Z",
    updated_at: "2026-05-12T15:30:00Z",
  },
  {
    id: 2,
    name: "运营工作区",
    code: "ops-workspace",
    description: "运营团队的工作区，负责日常运营任务",
    status: "active",
    org_id: 1,
    owner_id: 1,
    member_count: 8,
    project_count: 5,
    created_at: "2026-04-15T08:00:00Z",
    updated_at: "2026-05-10T12:00:00Z",
  },
  {
    id: 3,
    name: "已禁用工作区",
    code: "disabled-workspace",
    description: "这是一个已禁用的工作区示例",
    status: "disabled",
    org_id: 1,
    owner_id: 1,
    member_count: 3,
    project_count: 1,
    created_at: "2026-03-20T09:00:00Z",
    updated_at: "2026-05-08T18:00:00Z",
  },
];

// ============================================================
// Workspace Settings (detail view)
// ============================================================
export const mockWorkspaceSettings = {
  id: 1,
  name: "CRM 工作区",
  code: "crm-workspace",
  description: "客户关系管理团队的工作区",
  status: "active" as WorkspaceStatus,
  org_id: 1,
  owner_id: 1,
  member_count: 15,
  project_count: 8,
  created_at: "2026-05-01T10:00:00Z",
  updated_at: "2026-05-12T15:30:00Z",
};

File: neo/backend/src/app/schemas/workspace.py
"""Workspace schemas."""

from __future__ import annotations

from datetime import datetime
from enum import Enum
from typing import List

from pydantic import BaseModel, Field


class WorkspaceStatusEnum(str, Enum):
    """Workspace status enum."""

    ACTIVE = "active"
    DISABLED = "disabled"


class MemberRoleEnum(str, Enum):
    """Workspace member role enum."""

    OWNER = "owner"
    ADMIN = "admin"
    MEMBER = "member"
    GUEST = "guest"


# ==================== Workspace Schemas ====================


class WorkspaceBase(BaseModel):
    """Base schema for workspace."""

    name: str = Field(..., description="Workspace name", min_length=1, max_length=50)
    description: str | None = Field(
        None,
        description="Description",
        max_length=500,
    )


class WorkspaceCreate(WorkspaceBase):
    """Schema for creating workspace."""

    org_id: int = Field(..., description="Organization ID", gt=0)


class WorkspaceUpdate(BaseModel):
    """Schema for updating workspace."""

    name: str | None = Field(None, description="Workspace name", min_length=1, max_length=50)
    description: str | None = Field(None, description="Description", max_length=500)


class WorkspaceResponse(BaseModel):
    """Schema for workspace response."""

    id: int
    name: str
    code: str
    description: str | None = None
    status: WorkspaceStatusEnum
    org_id: int
    owner_id: int
    settings: str | None = None
    created_at: datetime
    updated_at: datetime
    disabled_at: datetime | None = None
    disabled_by: int | None = None
    # Computed fields
    member_count: int | None = None
    project_count: int | None = None

    model_config = {"from_attributes": True}


class WorkspaceListItem(BaseModel):
    """Schema for workspace list item."""

    id: int
    name: str
    code: str
    description: str | None = None
    status: WorkspaceStatusEnum
    org_id: int
    owner_id: int
    created_at: datetime
    updated_at: datetime
    # Computed fields
    member_count: int | None = None
    project_count: int | None = None

    model_config = {"from_attributes": True}


class WorkspaceListResponse(BaseModel):
    """Schema for workspace list response."""

    total: int
    page: int
    page_size: int
    list: List[WorkspaceListItem]


class WorkspaceStatusUpdate(BaseModel):
    """Schema for updating workspace status."""

    # No fields needed, uses endpoint path to determine action


class TransferOwnerRequest(BaseModel):
    """Schema for transferring workspace ownership."""

    new_owner_id: int = Field(..., description="New owner user ID", gt=0)


# ==================== Workspace Member Schemas ====================


class MemberBase(BaseModel):
    """Base schema for workspace member."""

    user_id: int = Field(..., description="User ID", gt=0)
    role: MemberRoleEnum = Field(..., description="Member role")


class MemberAdd(MemberBase):
    """Schema for adding workspace member."""


class MemberUpdate(BaseModel):
    """Schema for updating workspace member role."""

    role: MemberRoleEnum = Field(..., description="New role")


class MemberResponse(BaseModel):
    """Schema for workspace member response."""

    id: int
    workspace_id: int
    user_id: int
    role: MemberRoleEnum
    created_at: datetime
    updated_at: datetime
    # Joined user info
    username: str | None = None
    phone: str | None = None

    model_config = {"from_attributes": True}


class MemberListItem(BaseModel):
    """Schema for member list item."""

    id: int
    user_id: int
    username: str | None = None
    phone: str | None = None
    role: MemberRoleEnum
    joined_at: datetime

    model_config = {"from_attributes": True}


class MemberListResponse(BaseModel):
    """Schema for member list response."""

    total: int
    page: int
    page_size: int
    list: List[MemberListItem]


# ==================== Pagination Schema ====================


class PaginatedWorkspaceResponse(BaseModel):
    """Schema for paginated workspace list response."""

    total: int
    page: int
    page_size: int
